Replace an old jeditable rich text editor with TinyMCE

pjanik opened the pull request:
The general idea was to remove an ancient jeditable wysiwyg editor and replace with TinyMCE that was already used by ITSI. That involved:

1. Upgrade of TinyMCE version (slightly different look).
2. New React components used in authoring mode (text editor).
3. Refactoring of ITSI components and mixins, so the code can be shared between ITSI and regular authoring components.
4. Cleanup of various TinyMCE configs. E.g. ITSI, Tracked Questions, and so on. They seemed unnecessary. I've tried to reduce it to only two: regular and minimal. Minimal is used in places where the width is limited (questions edit dialogs). The basic one is pretty much an old ITSI config (and ofc still used by ITSI).
5. Jeditable lib was also to edit interactive page title in place. I've implemented simple React component that does the same, so Jeditable could be removed.
6. Jeditable lib removed.
7. A better approach to sidebar title editing. Now, when you click "edit", it will edit both title and content.
8. I've made sure that custom HTML doesn't break page and sidebar titles (we've seen that during demo). It will be escaped and html tags displayed, instead of being interpreted as before.

Screenshots:
1. Introduction:
    <img width="972" alt="screen shot 2018-04-03 at 12 21 37" src="https://user-images.githubusercontent.com/767857/38278722-09cfe73a-3752-11e8-94dc-96f5ee52d63f.png">
2. Sidebar editing:
    <img width="968" alt="screen shot 2018-04-03 at 12 21 52" src="https://user-images.githubusercontent.com/767857/38278735-19ed236c-3752-11e8-9bfe-4d57d6dad4b9.png">
3. Page title:
    <img width="424" alt="screen shot 2018-04-03 at 12 22 11" src="https://user-images.githubusercontent.com/767857/38278740-2143da16-3752-11e8-8997-823fd2822ad2.png">
4. Question edit dialog (minimal TinyMCE config):
    <img width="459" alt="screen shot 2018-04-03 at 15 06 49" src="https://user-images.githubusercontent.com/767857/38278752-34e3d0b2-3752-11e8-87b1-dc9424e44e4c.png">
5. Activity edit form (I made it a bit wider):
    <img width="660" alt="screen shot 2018-04-03 at 15 06 24" src="https://user-images.githubusercontent.com/767857/38278770-435e8e48-3752-11e8-85e0-bf797f05000a.png">

There are more places where TinyMCE is used, but these above are perhaps the most important.

scytacki commented on app/views/embeddable/open_responses/_form.html.haml:
```diff
@@ -15,9 +15,10 @@
+
+  :javascript
+    $(function() {
+      initTinyMCE('.wysiwyg', window.TinyMCEConfigMinimal);
```
It looks like this would apply the minimal config to all `.wysiwyg` elements on the page. What happens when there are some elements that should be minimal and some that should be basic?  Or are there never both in the DOM at the same time?

pjanik commented:
@scytacki, thanks, I've addressed two issues. I hoped ITSI would be okay with code editor too. And I've used different class/selector for minimal config (`wysiwyg-minimal`). I started with per-page initialization and added the general one, which uses `.wysiwyg` selector, later.
